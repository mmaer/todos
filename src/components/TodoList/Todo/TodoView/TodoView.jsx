import React from "react";
import classNames from "classnames";

import "./TodoView.scss";

const TodoView = ({
  todo,
  id,
  removeTodo,
  toggleCompletedStatus,
  toggleEditingStatus
}) => {
  const { completed, title } = todo;

  const toggleEditing = () => toggleEditingStatus(id);
  const toggleCompleted = () => toggleCompletedStatus(id);
  const remove = () => removeTodo(id);

  const labelClassNames = classNames("todo-view__label", {
    "todo-view__label--completed": completed
  });

  return (
    <div className="todo-view" onDoubleClick={toggleEditing}>
      <input
        type="checkbox"
        className="todo-view__toggle"
        checked={completed}
        onChange={toggleCompleted}
      />
      <label className={labelClassNames}>{title}</label>
      <button className="todo-view__remove" onClick={remove} />
    </div>
  );
};

export default TodoView;
