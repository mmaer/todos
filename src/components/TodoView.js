import React from "react";

const TodoView = ({
  todo,
  id,
  removeTodo,
  toggleCompletedStatus,
  toggleEditingStatus
}) => {
  const { completed, title } = todo;

  const toggleEditing = () => toggleEditingStatus(id);
  const toggleCompleted = () => toggleCompletedStatus(id);
  const remove = () => removeTodo(id);

  return (
    <div className="view" onDoubleClick={toggleEditing}>
      <input
        type="checkbox"
        className="toggle"
        checked={completed}
        onChange={toggleCompleted}
      />
      <label>{title}</label>
      <button className="destroy" onClick={remove} />
    </div>
  );
};

export default TodoView;
