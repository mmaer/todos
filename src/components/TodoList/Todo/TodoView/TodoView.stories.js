import React from "react";

import TodoView from ".";

export default {
  title: "TodoView"
};

const TODO = { title: "Todo", editing: false, completed: false, id: 1 };

export const todoView = () => <TodoView {...TODO} />;
