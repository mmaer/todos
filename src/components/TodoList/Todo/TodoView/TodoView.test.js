import React from "react";
import { shallow, mount } from "enzyme";

import TodoView from "./";

const TODO = {
  title: "title",
  completed: false,
  id: 1,
  removeTodo: () => {},
  toggleCompletedStatus: () => {},
  toggleEditingStatus: () => {}
};

describe("TodoView component", () => {
  it("Should match snapshot", () => {
    const wrapper = shallow(<TodoView {...TODO} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should match snapshot when todo is completed", () => {
    const wrapper = shallow(<TodoView {...TODO} completed />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should call toggleEditingStatus function when component is double clicked", () => {
    const mockToggleEditingFunc = jest.fn();
    const wrapper = mount(
      <TodoView {...TODO} toggleEditingStatus={mockToggleEditingFunc} />
    );
    wrapper.simulate("doubleclick");

    expect(mockToggleEditingFunc.mock.calls.length).toBe(1);
  });

  it("Should call toggleCompletedStatus function when checkbox is clicked", () => {
    const mockToggleCompletedStatusFunc = jest.fn();
    const wrapper = mount(
      <TodoView
        {...TODO}
        toggleCompletedStatus={mockToggleCompletedStatusFunc}
      />
    );
    const checkbox = wrapper.find("input");
    checkbox.simulate("change");

    expect(mockToggleCompletedStatusFunc.mock.calls.length).toBe(1);
  });

  it("Should call removeTodo function when remove button is clicked", () => {
    const mockRemoveTodoFunc = jest.fn();
    const wrapper = mount(
      <TodoView {...TODO} removeTodo={mockRemoveTodoFunc} />
    );
    const checkbox = wrapper.find("button");
    checkbox.simulate("click");

    expect(mockRemoveTodoFunc.mock.calls.length).toBe(1);
  });
});
