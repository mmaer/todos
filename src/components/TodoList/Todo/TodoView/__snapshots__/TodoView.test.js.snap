// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoView component Should match snapshot 1`] = `
<div
  className="todo-view"
  onDoubleClick={[Function]}
>
  <input
    checked={false}
    className="todo-view__toggle"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="todo-view__label"
  >
    title
  </label>
  <button
    className="todo-view__remove"
    onClick={[Function]}
  />
</div>
`;

exports[`TodoView component Should match snapshot when todo is completed 1`] = `
<div
  className="todo-view"
  onDoubleClick={[Function]}
>
  <input
    checked={true}
    className="todo-view__toggle"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="todo-view__label todo-view__label--completed"
  >
    title
  </label>
  <button
    className="todo-view__remove"
    onClick={[Function]}
  />
</div>
`;
